macros/APIListAlpha.ejs

```diff
@@ -65,74 +103,91 @@ function containsTag(tagList, tag) {
 
-for (var i = 0; i < numLetters; i++) {
-    var curLetter = letters[i];
-    var insertedHeading = false; // Haven't done this letter's heading yet
-    
     if (p < numPages) {
         do {
-            var tags = pages[p].tags;
-            var url = pages[p].url;
-            var summary = pages[p].summary;
-            var title = pages[p].title;
-            
+            const page = pages[p];
+            let builtTitle, badge = "";
+            let {
+                tags,
+                url,
+                summary = null,
+                title,
+                slug,
+            } = page;
+
             // Build the formatted title string; skip this page if it's not
             // an interface.
-            
-            var builtTitle = buildTitle(title);
-            if (builtTitle == null) {
+
+            slug = slug.replace('Web/API/', '');
+            if (!isInterface(slug, tags)) {
                 p++;
                 continue;
             }
 
-            var badge = "";
-            
+            [
+                builtTitle,
+                title,
+            ] = (buildTitle(title, slug));
+
             // add badges if needed
-            
+
             if (containsTag(tags, "Non-standard") || containsTag(tags, "Non standard")) {
                 badge = " " + await template("NonStandardBadge", ["1"]);
             }
-            
+
             if (containsTag(tags, "Obsolete")) {
                 badge += " " + await template("ObsoleteBadge", [1]);
             } else if (containsTag(tags, "Deprecated")) {
                 badge += " " + await template("DeprecatedBadge", [1]);
             }
-            
+
             if (containsTag(tags, "Experimental")) {
                 badge += " " + await template("ExperimentalBadge", ["1"]);
             }
-            
+
             // Wrap the badges in another span
-            
+
             if (badge.length) badge = '<span class="indexListBadges">' + badge + "</span>";
-                    
+
             if (title[0].toUpperCase() == curLetter) {
                 if (!containsTag(tags, "junk")) {
                     if (!insertedHeading) {
-                        html += "<span>" + curLetter + "</span><ul>";
+                        html += "\t<span>" + curLetter + "</span>\n\t<ul>";
                         insertedHeading = true;
                     }
-                    html += '\n<li><span class="indexListRow"><span class="indexListTerm"><a href="' + url.replace("en-US", locale) + '" title="' + wiki.escapeQuotes(summary) + '">' + builtTitle + '</a></span>' + badge + '</span></li>';
+                    html += '\n\t\t<li>' +
+                        '<span class="indexListRow">' +
+                            '<span class="indexListTerm">' +
+                                '<a href="' + url.replace("en-US", locale) + '"' +
+                                    (summary ? ` title="${mdn.escapeQuotes(summary)}"` : '') +
```

Comment:
This ensures that pages with an empty `summary` won’t have a pointless empty `title` attribute, which confuses some browsers (looking at you, **IE**/**Edge**).